pkg/chunk/cache/diskcache.go

```diff
@@ -10,14 +10,42 @@ import (
 	"sync"
 
 	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/tsdb/fileutil"
 	"golang.org/x/sys/unix"
 )
 
+var (
+	bucketsTotal = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace: "cortex",
+		Name:      "diskcache_buckets_total",
+		Help:      "Total count of buckets in the cache.",
+	})
+	bucketsInitialized = prometheus.NewCounter(prometheus.CounterOpts{
+		Namespace: "cortex",
+		Name:      "diskcache_buckets_initialized_total",
+		Help:      "total number of buckets that have been initialized to cache a chunk",
+	})
+	collisionsTotal = prometheus.NewCounter(prometheus.CounterOpts{
+		Namespace: "cortex",
+		Name:      "diskcache_collisions_total",
+		Help:      "total number of collisions when storing a key in the cache",
+	})
+)
+
+func init() {
+	prometheus.MustRegister(bucketsTotal)
+	prometheus.MustRegister(bucketsInitialized)
+	prometheus.MustRegister(collisionsTotal)
+}
+
 // TODO: in the future we could cuckoo hash or linear probe.
 
-// Buckets contain key (~50), chunks (1024) and their metadata (~100)
-const bucketSize = 2048
+// Buckets contain chunks (1024) and their metadata (~100)
+const (
+	bucketSize = 2048 // Buckets contain chunks (1024) and their metadata (~100)
```

Comment:
Hmm, `(1024) + 8bytes`? Maybe decrease the bucket size a little here?

Reply:
Metadata for a chunk can be much bigger than 8 bytes - its a json blob with all the labels in.